src/test/app/NFToken_test.cpp

```diff
@@ -6565,6 +6565,203 @@ class NFToken_test : public beast::unit_test::suite
         }
     }
 
+    void
+    testTxJsonMetaFields(FeatureBitset features)
+    {
+        // `nftoken_id` is added in the `tx` response for NFTokenMint and
+        // NFTokenAcceptOffer.
+        //
+        // `nftoken_ids` is added in the `tx` response for NFTokenCancelOffer
+        //
+        // `offer_id` is added in the `tx` response for NFTokenCreateOffer
+        //
+        // The values of these fields are dependent on the NFTokenID/OfferID
+        // changed in its corresponding transaction. We want to validate each
+        // transaction to make sure the synethic fields hold the right values.
+
+        testcase("Test synthetic fields from JSON response");
+
+        using namespace test::jtx;
+
+        Account const alice{"alice"};
+        Account const bob{"bob"};
+        Account const broker{"broker"};
+
+        Env env{*this, features};
+        env.fund(XRP(10000), alice, bob, broker);
+        env.close();
+
+        // Verify `nftoken_id` value equals to the NFTokenID that was
+        // changed in the most recent NFTokenMint or NFTokenAcceptOffer
+        // transaction
+        auto verifyNFTokenID = [&](uint256 const& actualNftID) {
+            // Get the hash for the most recent transaction.
+            std::string const txHash{
+                env.tx()->getJson(JsonOptions::none)[jss::hash].asString()};
+
+            env.close();
+            Json::Value const meta =
+                env.rpc("tx", txHash)[jss::result][jss::meta];
+
+            // Expect nftokens_id field
+            if (!BEAST_EXPECT(meta.isMember(jss::nftoken_id)))
+                return;
+
+            // Check the value of NFT ID in the meta with the
+            // actual value
+            uint256 nftID;
+            BEAST_EXPECT(nftID.parseHex(meta[jss::nftoken_id].asString()));
+            BEAST_EXPECT(nftID == actualNftID);
+        };
+
+        // Verify `nftoken_ids` value equals to the NFTokenIDs that were
+        // changed in the most recent NFTokenCancelOffer transaction
+        auto verifyNFTokenIDs = [&](std::vector<uint256> actualNftIDs) {
```

Comment:
You have lambdas named `verifyNFTokenID` and `verifyNFTokenIDs`.  But it turns out that they only work on specific transaction types.  At the least I'd like to see the name of this lambda changed to `verifyNFTokenIDsInCancelOffer()`.